common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -238,8 +243,78 @@ def student_view(self, context):
 
         fragment.add_content(self.system.render_template("seq_module.html", params))
 
+        self._capture_full_seq_item_metrics(display_items)
+        self._capture_current_unit_metrics(display_items)
+
+        # Get all descendant XBlock types and counts
         return fragment
 
+    def _locations_in_subtree(self, node):
+        """
+        The usage keys for all descendants of an XBlock/XModule as a flat list.
+        """
+        stack = [node]
+        locations = []
+
+        while stack:
+            curr = stack.pop()
+            locations.append(curr.location)
+            if curr.has_children:
+                stack.extend(curr.get_children())
```

Comment:
Are `has_children` and `get_children` performant enough to not cause a performance hit here? I'm thinking specifically of the ProblemBuilder XBlock that has nested XBlocks. Should be near the course leaves - but are there situations where a significant number of blocks/modules will exist beneath?